sdk/go/common/workspace/creds.go

```diff
@@ -213,5 +213,23 @@ func StoreCredentials(creds Credentials) error {
 	if err != nil {
 		return errors.Wrapf(err, "marshalling credentials object")
 	}
-	return ioutil.WriteFile(credsFile, raw, 0600)
+
+	tempCredsFile, err := ioutil.TempFile("", "credentials-*.json")
```

Comment:
Nit: Add a brief comment explaining why we write to a temporary file first and then rename (for concurrency safe writes) vs. just writing directly?